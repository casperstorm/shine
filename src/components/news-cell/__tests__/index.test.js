import renderer from 'react-test-renderer'
import React from 'react'

import NewsCell from '../index'

describe('news-cell', () => {
  const title = 'foo'
  const published = '2m'
  const currencies = [
    { code: 'BTC', title: 'BTC' },
    { code: 'VTC', title: 'VTC' },
    { code: 'GRS', title: 'GRS' },
  ]
  const votes = { important: 1, negative: 2, positive: 3 }

  it('should render correctly with all props', () => {
    const tree = renderer.create(
      <NewsCell
        title={title}
        published={published}
        currencies={currencies}
        votes={votes}
        onPress={() => {}}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
