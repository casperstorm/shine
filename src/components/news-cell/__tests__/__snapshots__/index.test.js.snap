// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`news-cell should render correctly with all props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "margin": 12,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Rubik-Regular",
          "fontSize": 17,
          "lineHeight": 22,
          "margin": 12,
          "marginBottom": 16,
        }
      }
    >
      foo
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 12,
            "marginLeft": 12,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 23,
                "justifyContent": "center",
                "marginRight": 7,
              },
              Object {
                "backgroundColor": "#419641",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Rubik-Bold",
                "fontSize": 12,
                "marginLeft": 8,
                "marginRight": 8,
              }
            }
          >
            VTC
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 23,
                "justifyContent": "center",
                "marginRight": 7,
              },
              Object {
                "backgroundColor": "#153c91",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Rubik-Bold",
                "fontSize": 12,
                "marginLeft": 8,
                "marginRight": 8,
              }
            }
          >
            BTC
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#252525",
              "borderRadius": 10,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 23,
              "justifyContent": "center",
              "marginRight": 7,
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": 8,
                "marginTop": 1,
              }
            }
          >
            <Image
              source={1}
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "marginLeft": 8,
                "marginRight": 8,
              }
            }
          >
            1
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#252525",
              "borderRadius": 10,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 23,
              "justifyContent": "center",
              "marginRight": 7,
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": 8,
                "marginTop": 1,
              }
            }
          >
            <Image
              source={1}
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "marginLeft": 8,
                "marginRight": 8,
              }
            }
          >
            1
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#252525",
              "borderRadius": 10,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 23,
              "justifyContent": "center",
              "marginRight": 7,
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": 8,
                "marginTop": 1,
              }
            }
          >
            <Image
              source={1}
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "marginLeft": 8,
                "marginRight": 8,
              }
            }
          >
            1
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Rubik-Light",
            "fontSize": 15,
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "opacity": 0.7,
          }
        }
      >
        2m
      </Text>
    </View>
    <View
      style={
        Object {
          "borderTopColor": "#444444",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "left": 12,
          "opacity": 0.8,
          "position": "absolute",
          "right": 12,
        }
      }
    />
  </View>
</View>
`;
